pyproject.toml

```diff
@@ -203,3 +168,53 @@ replace = "__version__ = '{new_version}'"
 [build-system]
 requires = ['setuptools', 'wheel']
 build-backend = 'setuptools.build_meta'
+
+[tool.ruff]
+preview = true
+line-length = 100
+indent-width = 4
+src = ["sdgym"]
+target-version = "py312"
```

Comment:
The [documentation says](https://docs.astral.sh/ruff/settings/#target-version) that this should be the minimum version. Should we change to 3.8?

Reply:
Good point. Given that ruff says to use `requires-python`, I just ended up removing it: https://github.com/sdv-dev/SDGym/pull/303/commits/bbdf1388fda6714bf1da48fbec664b78d50e0e4a
> If you're already using a pyproject.toml file, we recommend project.requires-python instead, as it's based on Python packaging standards, and will be respected by other tools. For example, Ruff treats the following as identical to target-version = "py38":